### DISCUSSION TITLE
go test keep failing

### DISCUSSION THREAD

**Discussion by RFCreate** [JanDeDobbeleer/oh-my-posh#2458] [Troubleshoot] [answered]
I have tried to test the application from source code, but it always throws me this error:

```
~$  go test -v ./...
# runtime/cgo
_cgo_export.c:3:20: fatal error: stdlib.h: No such file or directory
compilation terminated.
FAIL    oh-my-posh/color [build failed]
FAIL    oh-my-posh/console [build failed]
FAIL    oh-my-posh/engine [build failed]
FAIL    oh-my-posh/environment [build failed]
FAIL    oh-my-posh/properties [build failed]
FAIL    oh-my-posh/segments [build failed]
FAIL    oh-my-posh/shell [build failed]
FAIL    oh-my-posh/template [build failed]
```

I guess it has to do something with this line `_cgo_export.c:3:20: fatal error: stdlib.h: No such file or directory`. But I have not been able to solve it by searching in the internet. Please help!

**Comment by JanDeDobbeleer** [comment, marked as answer]
You're probably missing the glibc header files on your machine. To solve (on Ubuntu):

```
sudo apt-get install build-essential
```

More info [here](https://askubuntu.com/questions/158996/how-do-i-get-the-libc-development-libraries-for-ubuntu-12-04).

**Reply by RFCreate** [reply]
This work on my machine. Thanks!
